nvme-topology.c

```diff
@@ -426,7 +426,7 @@ static int legacy_list(struct nvme_topology *t)
 			continue;
 		ret = 0;
 
-		fd = open(path, O_RDONLY);
+		fd = open_global_device(path);
```

Comment:
We only want to report devices like '/dev/nvme0' which could be either directly attached nvme drive or SRIOV VF bound to current host with 'switchtec-nvme-cli list'. Are you trying to report 'pdfid@/dev/switchtecX' as well?

Reply:
Function `open_global_device` will check file name and open device accordingly, it also populates global_device with the right fops functions.

Reply:
It looks odd repeating open it without a close.